website/src/components/discourse/discourse.cy.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react'
+import { DiscourseFeed } from './index'
+
+const mockResponseData = [
+    {
+        "id": 4911,
+        "title": "Processing source tables with differing arrival times throughout day",
+        "has_accepted_answer": true,
+        "author": "MatthewMunn",
+        "like_count": 1,
+        "posts_count": 2
+    },
+]
+
+describe('Test Discourse component', () => {
+    it('Should render same text passed into title prop', () => {
+        cy.mount(<DiscourseFeed title='Open topics' />)
+        cy.contains('Open topics').should('be.visible')
+    })
+
+    it('Should display loading icon on inital load', () => {
+        cy.intercept('POST', 'http://localhost:8888/.netlify/functions/get-discourse-topics', {
+            delayMs: 10,
+        }
+        ).as('getTopics')
+        cy.mount(<DiscourseFeed />)
+        cy.get('[data-testid="feed-loader"]', { timeout: 100 }).should('be.visible')
+        cy.get('[data-testid="error-text"]').should('be.visible')
+
+    })
+
+    it('Should display Discourse data after API fetch', () => {
+        cy.intercept('POST', 'http://localhost:8888/.netlify/functions/get-discourse-topics', (req) => {
+            req.reply(mockResponseData);
+        }).as('getTopics')
+
+        cy.mount(<DiscourseFeed />)
+        cy.wait('@getTopics')
+        cy.get('[data-testid="topics-list"]').as('topicsList')
```

Comment:
This is really cool, didn't know about the `.as()` method!

Reply:
I know right I found it reading an article.

Reply:
Yeah `.as()` look like it will be very convenient!